src/main/scala/viper/gobra/translator/implementations/translator/AssertionsImpl.scala

```diff
@@ -44,8 +46,24 @@ class AssertionsImpl extends Assertions {
         acc.e match {
           case in.Accessible.Predicate(op) => ctx.predicate.predicateAccess(ctx)(op, acc.p)
           case in.Accessible.Address(op) => ctx.typeEncoding.addressFootprint(ctx)(op, acc.p)
-          case in.Accessible.ExprAccess(op) =>
-            op.typ match {
+          case in.Accessible.ExprAccess(op, underlyingType) =>
+            underlyingType match {
+              case _: in.SliceT =>
+                val iterVar = in.BoundVar(Names.freshName, in.IntT(Addressability.Exclusive))(acc.info)
+                val quantifiedAssert = in.SepForall(
+                  vars = Vector(iterVar),
+                  triggers = Vector(in.Trigger(Vector(in.IndexedExp(op, iterVar, underlyingType)(acc.info)))(acc.info)),
+                  body = in.Implication(
+                    in.And(
+                      in.AtMostCmp(in.IntLit(0)(acc.info), iterVar)(acc.info),
+                      in.LessCmp(iterVar, in.Length(op)(acc.info))(acc.info))(acc.info),
+                    in.Access(
+                      in.Accessible.Address(in.IndexedExp(op, iterVar, underlyingType)(acc.info)),
+                      acc.p
+                    )(acc.info)
+                  )(acc.info)
+                )(acc.info)
+                ctx.ass.translate(quantifiedAssert)(ctx)
```

Comment:
I would move this into the slice encoding (like the case for the map is in the map encoding).